src/functions/index.ts

```diff
@@ -74,12 +75,43 @@ class FunctionRegistry extends Registry<FunctionDescription> {
     }
     const descr = addMetaInfoFromArg(addDescr);
     validateArguments(descr.args);
-    this.mapping[name] = createComputeFunctionFromDescription(descr);
+    this.mapping[name] = addInputHandling(descr, createComputeFunctionFromDescription(descr));
     super.add(name, descr);
     return this;
   }
 }
 
+function addInputHandling(
+  descr: FunctionDescription,
+  computeFunction: ComputeFunction<Arg, Matrix<ValueAndFormat> | ValueAndFormat>
+) {
+  return function (this: EvalContext, ...args: Arg[]) {
+    const checkedArgs: Arg[] = [];
+    for (let i = 0; i < args.length; i++) {
+      const argDefinition = descr.args[descr.getArgToFocus(i + 1) - 1];
+      const acceptMatrix = argDefinition.type.some((t) => t.startsWith("RANGE"));
+      const arg = args[i];
+      if (isMatrix(arg) && !acceptMatrix) {
+        if (arg.length !== 1 || arg[0].length !== 1) {
+          throw new EvaluationError(
+            CellErrorType.GenericError,
+            _t(
+              "Function [[FUNCTION_NAME]] expects the parameter '%s' to be a single value or a single cell reference, not a range.",
+              argDefinition.name
+            )
+          );
+        }
+        checkedArgs.push(arg[0][0]);
+      } else {
+        checkedArgs.push(arg);
+      }
+    }
+
+    const compute = computeFunction.bind(this);
+    return compute(...checkedArgs);
```

Comment:
I believe this is faster since it doesn't create a new (binded) function and avoid the array destructuring.
```suggestion
    return computeFunction.apply(this, args);
```

Reply:
probably worth using `.apply` in `buildComputeFunctionFromDescription` as well. We should measure the impact